Automattic/jetpack
# E2E tests: Slack notifications

#### Changes proposed in this Pull Request:

This PR improves the Slack notifications for E2E tests. The scope was to produce less noise while enhancing the provided information for a quick evaluation.

Changes:
* Send notification for each failed test run (matrix group).
* Send a single notification for each successful E2E workflow run. This is more as a heartbeat notification, in case there are no test failures we still want to know the setup still works fine.
* Included the test suite name.
* Included list of failed tests.
* Build notification content out of Jest's test summary json file instead of custom slack log file.
* Log files are no longer uploaded in Slack, only stacktraces and screenshots are, for a quick evaluation. In case more details are needed we can check the logs together with the full test output as they are attached in Github.
* General look and feel.

#### Jetpack product discussion
1200222342137241-as-1200182606016954

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* Locally, you can set `GITHUB_EVENT_PATH` and `GITHUB_RUN_ID` env variables and run `./bin/slack.js` as per instructions.
* In CI, check the notifications channel (example: p1620640477371800-slack-C1Y1F7ZM1)
